### DISCUSSION TITLE
Difficulties adding libinput-gestures to hyprland on Fedora Atomic

### DISCUSSION THREAD

**Discussion by mecattaf** [bulletmark/libinput-gestures#362] [General]
```
libinput-gestures: session Hyprland+wayland on Linux-6.8.9-300.fc40.x86_64-x86_64-with-glibc2.39, python 3.12.3, libinput 1.25.0
Hash: 3766cbfdefd99f715a7f1ed5ad4f5da7
Gestures configured in ~/.config/libinput-gestures.conf:
swipe right      3 /var/home/dev/.config/hypr/scripts/gestures backward
swipe left       3 /var/home/dev/.config/hypr/scripts/gestures forward
swipe up         3 hyprctl dispatch fullscreen 1
swipe down       3 hyprctl dispatch fullscreen 1
swipe right      4 hyprctl dispatch layoutmsg cyclenext
swipe left       4 hyprctl dispatch layoutmsg cycleprev
pinch out        4 swaymsg gaps inner current plus 50
pinch in         4 swaymsg gaps inner current minus 50
pinch out        5 swaymsg gaps inner current plus 50
pinch in         5 swaymsg gaps inner current minus 50
Failed to open /dev/input/event2 (Permission denied)
Failed to open /dev/input/event18 (Permission denied)
Failed to open /dev/input/event0 (Permission denied)
Failed to open /dev/input/event1 (Permission denied)
Failed to open /dev/input/event9 (Permission denied)
Failed to open /dev/input/event10 (Permission denied)
Failed to open /dev/input/event16 (Permission denied)
Failed to open /dev/input/event17 (Permission denied)
Failed to open /dev/input/event11 (Permission denied)
Failed to open /dev/input/event12 (Permission denied)
Failed to open /dev/input/event13 (Permission denied)
Failed to open /dev/input/event14 (Permission denied)
Failed to open /dev/input/event15 (Permission denied)
Failed to open /dev/input/event4 (Permission denied)
Failed to open /dev/input/event5 (Permission denied)
Failed to open /dev/input/event6 (Permission denied)
Failed to open /dev/input/event7 (Permission denied)
Failed to open /dev/input/event8 (Permission denied)
Failed to open /dev/input/event22 (Permission denied)
Failed to open /dev/input/event23 (Permission denied)
Failed to open /dev/input/event24 (Permission denied)
Failed to open /dev/input/event25 (Permission denied)
Failed to open /dev/input/event19 (Permission denied)
Failed to open /dev/input/event21 (Permission denied)
Failed to open /dev/input/event3 (Permission denied)
Failed to open /dev/input/event20 (Permission denied)
Can not see any devices, did you add yourself to the input group and reboot?
Could not determine touchpad device.
```

**Describe the issue**
Hi all,
I am attempting to add libinput-gestures to my Fedora Atomic 40 image, which runs hyprland window manager. 
My current approach is described at length here: https://github.com/wayblueorg/wayblue/issues/42

Hoping somebody might have a solution :)

**Comment by bulletmark** [comment]
Given you are asking for configuration help I have converted this from an issue to a discussion.

Please paste the output of `ls -l /dev/input`, and `groups` for that user.

**Comment by mecattaf** [comment]
My user name is "dev"

~ 
❯ groups dev
dev : dev wheel

~ 
❯ ls -l /dev/input/
total 0
drwxr-xr-x. 2 root root     160 May 15 16:42 by-id/
drwxr-xr-x. 2 root root     500 May 15 16:42 by-path/
crw-rw----. 1 root input 13, 64 May 15 16:42 event0
crw-rw----. 1 root input 13, 65 May 15 16:42 event1
crw-rw----. 1 root input 13, 74 May 15 16:42 event10
crw-rw----. 1 root input 13, 75 May 15 16:42 event11
crw-rw----. 1 root input 13, 76 May 15 16:42 event12
crw-rw----. 1 root input 13, 77 May 15 16:42 event13
crw-rw----. 1 root input 13, 78 May 15 16:42 event14
crw-rw----. 1 root input 13, 79 May 15 16:42 event15
crw-rw----. 1 root input 13, 80 May 15 16:42 event16
crw-rw----. 1 root input 13, 81 May 15 16:42 event17
crw-rw----. 1 root input 13, 82 May 15 16:42 event18
crw-rw----. 1 root input 13, 83 May 15 16:42 event19
crw-rw----. 1 root input 13, 66 May 15 16:42 event2
crw-rw----. 1 root input 13, 84 May 15 16:42 event20
crw-rw----. 1 root input 13, 85 May 15 16:42 event21
crw-rw----. 1 root input 13, 86 May 15 16:42 event22
crw-rw----. 1 root input 13, 87 May 15 16:42 event23
crw-rw----. 1 root input 13, 88 May 15 16:42 event24
crw-rw----. 1 root input 13, 89 May 15 16:42 event25
crw-rw----. 1 root input 13, 67 May 15 16:42 event3
crw-rw----. 1 root input 13, 68 May 15 16:42 event4
crw-rw----. 1 root input 13, 69 May 15 16:42 event5
crw-rw----. 1 root input 13, 70 May 15 16:42 event6
crw-rw----. 1 root input 13, 71 May 15 16:42 event7
crw-rw----. 1 root input 13, 72 May 15 16:42 event8
crw-rw----. 1 root input 13, 73 May 15 16:42 event9
crw-rw----. 1 root input 13, 63 May 15 16:42 mice
crw-rw----. 1 root input 13, 32 May 15 16:42 mouse0
crw-rw----. 1 root input 13, 33 May 15 16:42 mouse1
crw-rw----. 1 root input 13, 34 May 15 16:42 mouse2
crw-rw----. 1 root input 13, 35 May 15 16:42 mouse3
crw-rw----. 1 root input 13, 36 May 15 16:42 mouse4
crw-rw----. 1 root input 13, 37 May 15 16:42 mouse5
crw-rw----. 1 root input 13, 38 May 15 16:42 mouse6
crw-rw----. 1 root input 13, 39 May 15 16:42 mouse7

folder structure inside /dev/input

by-id
│  usb-Apple_Inc._Magic_Keyboard_F0T2502RPN512HXAZ-event-if00 ➛ event11
│  usb-Apple_Inc._Magic_Keyboard_F0T2502RPN512HXAZ-if01-event-kbd ➛ event12
│  usb-Apple_Inc._Magic_Trackpad_CC2333402XA27CRAN-event-mouse ➛ event4
│  usb-Apple_Inc._Magic_Trackpad_CC2333402XA27CRAN-if01-event-mouse ➛ event5
│  usb-Apple_Inc._Magic_Trackpad_CC2333402XA27CRAN-if01-mouse ➛ mouse1
└  usb-Apple_Inc._Magic_Trackpad_CC2333402XA27CRAN-mouse ➛ mouse0
 by-path
│  pci-0000:00:14.0-usb-0:1.3:1.0-event-mouse ➛ event4
│  pci-0000:00:14.0-usb-0:1.3:1.0-mouse ➛ mouse0
│  pci-0000:00:14.0-usb-0:1.3:1.1-event-mouse ➛ event5
│  pci-0000:00:14.0-usb-0:1.3:1.1-mouse ➛ mouse1
│  pci-0000:00:14.0-usb-0:1.4:1.0-event ➛ event11
│  pci-0000:00:14.0-usb-0:1.4:1.1-event-kbd ➛ event12
│  pci-0000:00:14.0-usbv2-0:1.3:1.0-event-mouse ➛ event4
│  pci-0000:00:14.0-usbv2-0:1.3:1.0-mouse ➛ mouse0
│  pci-0000:00:14.0-usbv2-0:1.3:1.1-event-mouse ➛ event5
│  pci-0000:00:14.0-usbv2-0:1.3:1.1-mouse ➛ mouse1
│  pci-0000:00:14.0-usbv2-0:1.4:1.0-event ➛ event11
│  pci-0000:00:14.0-usbv2-0:1.4:1.1-event-kbd ➛ event12
│  pci-0000:00:15.0-platform-i2c_designware.0-event ➛ event13
│  pci-0000:00:15.0-platform-i2c_designware.0-event-mouse ➛ event17
│  pci-0000:00:15.0-platform-i2c_designware.0-mouse ➛ mouse7
│  pci-0000:00:19.1-platform-i2c_designware.3-event ➛ event6
│  pci-0000:00:19.1-platform-i2c_designware.3-event-mouse ➛ event10
│  pci-0000:00:19.1-platform-i2c_designware.3-mouse ➛ mouse4
│  pci-0000:00:1f.3-platform-skl_hda_dsp_generic-event ➛ event22
│  platform-asus-nb-wmi-event ➛ event21
│  platform-i8042-serio-0-event-kbd ➛ event3
│  platform-INTC1077:00-event ➛ event19
└  platform-pcspkr-event-spkr

Note that I am on Fedora Atomic 40

**Comment by bulletmark** [comment]
Note your first output above shows you are not a member of the "input" group and note the error printed in the second last line of the first error output you pasted here.

**Reply by mecattaf** [reply]
I just ran `sudo usermod -aG input dev` but the output of 
```
groups dev
```
remains
dev : dev wheel

I can confirm that the input group does exist: `getent group input` returns input:x:104:

Is there another way to make the "dev" user a member of "input" group?

**Reply by elvisisvan** [reply]
`reboot`

**Comment by bulletmark** [comment]
That same message tells you to reboot. Did you do that?
